File: core/src/main/java/org/dockbox/darwin/core/events/AbstractTargetEvent.java
package org.dockbox.darwin.core.events;

import org.dockbox.darwin.core.objects.events.Targetable;
import org.dockbox.darwin.core.objects.targets.Target;
import org.jetbrains.annotations.NotNull;

public abstract class AbstractTargetEvent implements Targetable {

    private Target target;

    public AbstractTargetEvent(Target target) {
        this.target = target;
    }

    @NotNull
    @Override
    public Target getTarget() {
        return this.target;
    }

    @Override
    public void setTarget(@NotNull Target target) {
        this.target = target;
    }
}

File: core/src/main/kotlin/org/dockbox/darwin/core/events/AbstractTargetEvent.kt
package org.dockbox.darwin.core.events

import org.dockbox.darwin.core.objects.events.Targetable
import org.dockbox.darwin.core.objects.targets.Target

abstract class AbstractTargetEvent(private var target: Target) : Targetable {
    override fun getTarget(): Target {
        return target
    }

    override fun setTarget(target: Target) {
        this.target = target
    }

}
